src/ott/solvers/linear/discrete_barycenter.py

```diff
@@ -33,67 +34,65 @@ class SinkhornBarycenterOutput(NamedTuple):  # noqa: D101
-    lse_mode: bool = True,
-    debiased: bool = False
-) -> SinkhornBarycenterOutput:
-  """Compute discrete barycenter :cite:`janati:20a`.
-
-  Args:
-    geom: geometry object.
-    a: batch of histograms of shape ``[batch, num_a]``.
-    weights: positive weights in the probability simplex.
-    dual_initialization: array of shape ``[batch, num_b]`` for the
-      initialization of `g_v`.
-    threshold: tolerance to monitor convergence.
-    norm_error: power used to define p-norm of error for marginal/target.
-    inner_iterations: the Sinkhorn error is not recomputed at each
-     iteration but every inner_num_iter instead to avoid computational overhead.
-    min_iterations: the minimum number of Sinkhorn iterations carried
-     out before the error is computed and monitored.
-    max_iterations: the maximum number of Sinkhorn iterations.
-    lse_mode: True for log-sum-exp computations, False for kernel multiply.
-    debiased: whether to run the debiased version of the Sinkhorn divergence.
-
-  Returns:
-    A ``SinkhornBarycenterOutput``, which contains two arrays of potentials,
-    each of size ``batch`` times ``geom.num_a``, summarizing the OT between each
-    histogram in the database onto the barycenter, described in ``histogram``,
-    as well as a sequence of errors that monitors convergence.
-  """
-  batch_size, num_a = a.shape
-  _, num_b = geom.shape
-
-  if weights is None:
-    weights = jnp.ones((batch_size,)) / batch_size
-  if weights.shape[0] != batch_size:
-    raise ValueError(f'weights must have size `{batch_size}`.')
-
-  if dual_initialization is None:
-    # initialization strategy from https://arxiv.org/pdf/1503.02533.pdf, (3.6)
-    dual_initialization = geom.apply_cost(a.T, axis=0).T
-    dual_initialization -= jnp.average(
-        dual_initialization, weights=weights, axis=0
-    )[jnp.newaxis, :]
-
-  if debiased and not geom.is_symmetric:
-    raise ValueError('Geometry must be symmetric to use debiased option.')
-  norm_error = (norm_error,)
-  return _discrete_barycenter(
-      geom, a, weights, dual_initialization, threshold, norm_error,
-      inner_iterations, min_iterations, max_iterations, lse_mode, debiased,
-      num_a, num_b
-  )
+@jax.tree_util.register_pytree_node_class
+class FixedBarycenter:
+
+  def __init__(
+      self,
+      threshold: float = 1e-2,
+      norm_error: int = 1,
+      inner_iterations: float = 10,
+      min_iterations: int = 0,
+      max_iterations: int = 2000,
+      lse_mode: bool = True,
+      debiased: bool = False
+  ):
+
+    self.threshold = threshold
+    self.norm_error = norm_error
+    self.inner_iterations = inner_iterations
+    self.min_iterations = min_iterations
+    self.max_iterations = max_iterations
+    self.lse_mode = lse_mode
+    self.debiased = debiased
+
+  def __call__(
+      self,
+      fixed_bp: barycenter_problem.FixedBarycenterProblem,
+      dual_initialization: Optional[jnp.ndarray] = None,
+  ) -> SinkhornBarycenterOutput:
```

Comment:
Please document this function (after #315 , linter will not pass).

Reply:
Thanks!